atom/browser/native_window_mac.mm

```diff
@@ -1205,6 +1211,67 @@ static bool FromV8(v8::Isolate* isolate, v8::Handle<v8::Value> val,
   return collectionBehavior & NSWindowCollectionBehaviorCanJoinAllSpaces;
 }
 
+void NativeWindowMac::SetVibrancy(const std::string& type) {
+  if (!(base::mac::IsOSMavericks() || base::mac::IsOSYosemiteOrLater())) return;
+
+  NSView* vibrant_view = [window_ vibrantView];
+
+  if (type.empty()) {
+    if (vibrant_view == nil) return;
+
+    [vibrant_view removeFromSuperview];
+    [window_ setVibrantView:nil];
+
+    return;
+  }
+
+  NSVisualEffectView* effect_view = (NSVisualEffectView*)vibrant_view;
+  if (effect_view == nil) {
+    effect_view = [[NSVisualEffectView alloc] initWithFrame:
```

Comment:
> and ARC will destroy it, when we set it to nil .

I don't believe Electron is compiled with ARC enabled. And it does look like `dealloc` is never called when the view is set to `nil`. Here are the changes I made to verify that:

```diff
diff --git a/atom/browser/native_window_mac.mm b/atom/browser/native_window_mac.mm
index a4633db..253608f 100644
--- a/atom/browser/native_window_mac.mm
+++ b/atom/browser/native_window_mac.mm
@@ -50,6 +50,16 @@
 @interface FullSizeContentView : NSView
 @end
 
+@interface MyView : NSVisualEffectView
+@end
+
+@implementation MyView
+  - (void)dealloc {
+    NSLog(@"dealloc called");
+    [super dealloc];
+  }
+@end
+
 @implementation FullSizeContentView
 
 // This method is directly called by NSWindow during a window resize on OSX
@@ -1227,7 +1237,7 @@ static bool FromV8(v8::Isolate* isolate, v8::Handle<v8::Value> val,
 
   NSVisualEffectView* effect_view = (NSVisualEffectView*)vibrant_view;
   if (effect_view == nil) {
-    effect_view = [[NSVisualEffectView alloc] initWithFrame:
+    effect_view = [[MyView alloc] initWithFrame:
       [[window_ contentView] bounds]];
     [window_ setVibrantView:(NSView*)effect_view];
```

Then ran the following from the Console of the dev tools:

```js
require('electron').remote.getCurrentWindow().setVibrancy('light');
require('electron').remote.getCurrentWindow().setVibrancy('');
```

`dealloc called` was never logged to the Console.

But if I added:

```objective-c
[effect_view autorelease];
```

And re-ran it, I did see `dealloc called` get logged when the vibrancy was removed.